[docs] Migrate Plugin contribution main page (#694)
moodle/devdocs: general/community/plugincontribution/index.md

1. We do not automatically pull from Github. You have to explicitly release new version via the Plugins directory. Still you are encouraged to use Git tags - releasing new version is easier then.

### Repository

- Git is de-facto standard source code management system for Moodle plugins.
- You are supposed to publish and share your plugin code so that others can clone it easily and eventually contribute patches for it easily.
- [Github](https://github.com/) is a popular choice of many Moodle developers these days.
- The layout should be that the root of the repository is the root of the plugin folder. In other words, in the root of your repository there should be files like version.php and folders like lang or classes. That way, your plugin can be directly cloned (checked out) into existing Moodle installation without conflicting with other contributed plugins of the same type, also obtained via Git.

> - It is suggested to follow the common naming convention of Moodle related repositories: moodle-{plugintype}_{pluginname}. For example, the birthday block has a repository name of moodle-block_birthday and is located at https://github.com/arborrow/moodle-block_birthday. Other developers can fork the code and work from their repositories.

Reviewer: ```suggestion
- It is suggested to follow the common naming convention of Moodle related repositories: moodle-`{plugintype}_{pluginname}`. For example, the birthday block has a repository name of `moodle-block_birthday` and is located at https://github.com/arborrow/moodle-block_birthday. Other developers can fork the code and work from their repositories.
```

### Tracker

You are expected to have a system where users can report feature requests, bugs, and other code issues. Following the open source best practise, these reports should be open and public and the associated procedures should be as transparent as possible.

> There are multiple options on where to host the issues tracker. Most plugin developers today prefer to have the issue tracker close to their source code management system - so they use the native issue tracker built into Github, BitBucket or some similar system. It is suggested to use a system that your users are likely to be familiar with.

Reviewer: ```suggestion
:::info

There are multiple options on where to host the issues tracker. Most plugin developers today prefer to have the issue tracker close to their source code management system - so they use the native issue tracker built into Github, BitBucket or some similar system. It is suggested to use a system that your users are likely to be familiar with.

:::
```

## Tutorial

There is a [Tutorial](https://docs.moodle.org/dev/Tutorial) to help you learn how to write plugins for Moodle from start to finish, while showing you how to navigate the most important developer documentation along the way.

## See also

- [Moodle icons](https://docs.moodle.org/dev/Moodle_icons)
- [Overview](../../community/intro.md) of Moodle development
- Using Moodle [Best practices for code modification?](http://moodle.org/mod/forum/discuss.php?d=99037) forum discussion

> - [Moodle](http://aosabook.org/en/moodle.html) in The Architecture of Open Source Applications, by Tim Hunt - an overview of all aspects of how Moodle works, it focuses on those areas where Moodle's design is particularly interesting.

Reviewer: ```suggestion
- [Moodle in The Architecture of Open Source Applications](http://aosabook.org/en/moodle.html), by Tim Hunt - an overview of all aspects of how Moodle works, it focuses on those areas where Moodle's design is particularly interesting.
```

> - [The Moodle 3.0 database schema](http://www.examulator.com/er/) reverse engineered and [Database schema introduction](https://docs.moodle.org/dev/Database_schema_introduction).

Reviewer: ```suggestion
- [The Moodle 3.0 database schema](https://moodledev.io/docs/apis/core/dml/database-schema) reverse engineered and [Database schema introduction](https://docs.moodle.org/dev/Database_schema_introduction).
```
